projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -3835,26 +3832,14 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
     }
 
     private createComponentInstance(component: any) {
-        // TODO: create component instance view viewContainerRef.createComponent(Component, Settings) overload
-        let dynamicFactory: ComponentFactory<any>;
-        const factoryResolver = this.moduleRef
-            ? this.moduleRef.componentFactoryResolver
-            : this.resolver;
-        try {
-            dynamicFactory = factoryResolver.resolveComponentFactory(component);
-        } catch (error) {
-            console.error(error);
-            return null;
-        }
-
         const injector = this.moduleRef
-            ? this.moduleRef.injector
-            : this.injector;
-        const dynamicComponent: ComponentRef<any> = dynamicFactory.create(
-            injector
-        );
-        this.appRef.attachView(dynamicComponent.hostView);
-
+        ? this.moduleRef.injector
+        : this.injector;
+        const dynamicComponent: ComponentRef<any> = this.viewRef.createComponent(component, { injector: injector } );
+        // move to app ref's views.
+        const index = this.viewRef.indexOf(dynamicComponent.hostView);
+        const viewRef = this.viewRef.detach(index);
+        this.appRef.attachView(viewRef);
```

Comment:
I did try to simplify this, however just creating it directly in any viewref caused the filter to not position correctly over the column for which it opens:
![positionIssue](https://user-images.githubusercontent.com/10397980/230881653-cad88e59-132f-41f9-898a-0b9863f84a89.gif)
Not sure why that's the case. Might be some issue with the overlay.

Detaching it and moving it to the old place where it was attached (in the app ref) fixed this and the previously failing tests - > https://github.com/IgniteUI/igniteui-angular/runs/12540307179.

Reply:
That's odd, since the actual overlay part is moved in the DOM regardless (to the grid outlet presumably), so it shouldn't affect the positioning in theory.
Regardless, this 'util' method in the grid base exists entirely to create the filter dialog before attaching it to the overlay service, when the service already has built-in logic to do the same thing that should likely be used instead. How about we accept this for now (matching the old logic) and log the refactor of the filter dialog separately as it's put of scope for this PR anyway.